pkg/collector/corechecks/snmp/snmp.go

```diff
@@ -49,19 +49,34 @@ type Check struct {
 	sessionFactory             session.Factory
 	workerRunDeviceCheckErrors *atomic.Uint64
 	agentConfig                config.Component
+	pythonCheck                check.Check
 }
 
 // Run executes the check
 func (c *Check) Run() error {
 	var checkErr error
 	sender, err := c.GetSender()
 	if err != nil {
-		return err
+		return c.RunPythonCheck(err)
```

Comment:
If the sender fails, would it be possible to get it in Python? I recall it uses the same sender manager

Reply:
We do  call `GetSender` within the Go <-> Python bridge:

https://github.com/DataDog/datadog-agent/blob/3ba0257d8eadd13d7dcf0b5616751a2aa5042e90/pkg/collector/python/aggregator.go#L31-L68

We call this code from the Python integration via the the `aggregator.submit_metric` 

https://github.com/DataDog/integrations-core/blob/98186cfdf5231a4efab3fbd05cae8f105da5430c/datadog_checks_base/datadog_checks/base/checks/base.py#L768